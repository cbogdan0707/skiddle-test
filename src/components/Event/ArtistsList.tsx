import React from 'react';
import { Link } from 'react-router-dom';
import { ArtistsData, EventData } from '../../types/types';

export const ArtistsList = ({
  response,
}: {
  response: { results: EventData };
}) => {
  return (
    <section>
      {response.results.artists && response.results.artists.length !== 0 && (
        <div className="skiddle-u-color-primary skiddle-u-mb-medium skiddle-c-artist-title">
          List of artists:
        </div>
      )}
      <ul className="skiddle-c-search-results-container">
        {response.results.artists &&
          response.results.artists.map((artist: ArtistsData) => (
            <li key={artist.artistid}>
              <Link
                className="skiddle-u-primary-button skiddle-u-pa-small"
                to={{ pathname: `/artist/${artist.artistid}` }}
              >
                {artist.name}
              </Link>
            </li>
          ))}
      </ul>
    </section>
  );
};
