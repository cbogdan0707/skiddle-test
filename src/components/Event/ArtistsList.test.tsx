import React from 'react';
import { render } from '@testing-library/react';
import { ArtistsList } from './ArtistsList';
import { BrowserRouter } from 'react-router-dom';

const mockResponse = {
  results: {
    artists: [
      {
        artistid: 123540290,
        name: 'Palms Trax',
        image:
          'https://i.scdn.co/image/ab67616d0000b273c22e0b4198f2f563566708ff',
      },
      {
        artistid: 123551074,
        name: 'High Hoops DJs',
        image:
          'https://i.scdn.co/image/ba6a30eefb09cbae4e1d9c253d0fe0afef5ba954',
      },
    ],
  },
};

describe('Artist List', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ArtistsList response={mockResponse} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
