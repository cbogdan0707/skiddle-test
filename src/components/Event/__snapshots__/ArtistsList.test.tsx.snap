// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist List should match snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="skiddle-u-color-primary skiddle-u-mb-medium skiddle-c-artist-title"
    >
      List of artists:
    </div>
    <ul
      class="skiddle-c-search-results-container"
    >
      <li>
        <a
          class="skiddle-u-primary-button skiddle-u-pa-small"
          href="/artist/123540290"
        >
          Palms Trax
        </a>
      </li>
      <li>
        <a
          class="skiddle-u-primary-button skiddle-u-pa-small"
          href="/artist/123551074"
        >
          High Hoops DJs
        </a>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
